examples/cache/verify.sh

```diff
@@ -68,6 +68,9 @@ check_cached "$response"
 run_log "Snooze for 3 more seconds"
 sleep 3
 
+run_log "Snooze for one more minute"
+sleep 60
```

Comment:
we removed/changed this previously and reduced the timings - preferably we dont want to add back a 1 minute sleep if possible

Reply:
The `valid-for-minute` response includes the `cache-control: max-age=60` header, which causes Envoy to serve the cached response without contacting the backend service for one minute. 
The `verify.sh` script verifies that after one minute has elapsed, Envoy validates the response with the backend service. It does this by checking if the `date` header in the response is later than the generation timestamp.

https://github.com/envoyproxy/envoy/blob/main/examples/cache/responses.yaml:
```
valid-for-minute:
  body: This response will stay fresh for one minute
  headers:
    cache-control: max-age=60
    custom-header: any value
```

The 60-second duration can be decreased (for instance, to 10 seconds). However, it's important to ensure that the documentation is also updated to reflect this change.

https://github.com/envoyproxy/envoy/blob/main/docs/root/start/sandboxes/cache.rst#1-valid-for-minute

Reply:
Please ignore my previous message. I failed to notice that the `ci-responses.yaml` file, which is utilized in the CI, has the maximum age set to 4 seconds (`cache-control: max-age=4`).

The `check_validated` function in `verify.sh` was passing even when the envoy proxy returned a 503 error. This was because the function only verified the absence of an `age` header and the inequality of the two extracted timestamps. Since there was no timestamp in the response body, the second timestamp was empty, causing the test to pass.

```
HTTP/1.1 503 Service Unavailable
content-length: 95
content-type: text/plain
date: Fri, 20 Oct 2023 04:08:45 GMT
server: envoy

upstream connect error or disconnect/reset before headers. reset reason: connection termination
```
To address this issue, I added a check for the HTTP response code.